pkg/devconsole/tekton-results/results.go

```diff
@@ -0,0 +1,299 @@
+		if err.Error() != addrError.Error() {
+			return false, err
+		}
+		// host string without any port specified
+		h = host
+	} else if len(p) == 0 {
+		return false, errors.New("invalid host name format")
+	}
+
+	// use ipv4 dotted decimal for further checking
+	if h == "localhost" {
+		h = "127.0.0.1"
+	}
+	ip := net.ParseIP(h)
+
+	return ip.IsLoopback(), nil
+}
+
+func getTRHost(r *http.Request, dynamicClient *dynamic.DynamicClient) (string, error) {
+	// If running on localhost, use the route to get the host.
+	// Required for local development.
+	localHost, err := isLocalhost(r)
+	if err != nil {
+		return "", err
+	}
+	if localHost {
+		route, err := dynamicClient.Resource(*TektonResultsAPIRoute).Namespace("openshift-pipelines").Get(context.TODO(), "tekton-results-api-service", metav1.GetOptions{})
+		if err != nil {
+			return "", err
+		}
+		cachedTektonResultsHost, isHostPresent, err := unstructured.NestedString(route.Object, "spec", "host")
+		if err != nil || !isHostPresent {
+			return "", err
+		}
+		return cachedTektonResultsHost, nil
+	}
+
+	if cachedTektonResultsHost != "" {
+		return cachedTektonResultsHost, nil
+	}
+
+	host, err := dynamicClient.Resource(*TektonResultsResource).Get(context.TODO(), "result", metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	targetNamespace, isTargetNsPresent, err := unstructured.NestedString(host.Object, "spec", "targetNamespace")
+	if err != nil || !isTargetNsPresent {
+		targetNamespace = ""
+	}
+	serverPort, isPortPresent, err := unstructured.NestedString(host.Object, "spec", "server_port")
+	if err != nil || !isPortPresent {
+		serverPort = "8080"
+	}
+	tlsHostname, isTLSHostnamePresent, err := unstructured.NestedString(host.Object, "spec", "tls_hostname_override")
+	if err != nil || !isTLSHostnamePresent {
+		tlsHostname = ""
+	}
+
+	if tlsHostname != "" {
+		cachedTektonResultsHost = fmt.Sprintf("%s:%s", tlsHostname, serverPort)
+	} else if targetNamespace != "" && serverPort != "" {
+		cachedTektonResultsHost = fmt.Sprintf("tekton-results-api-service.%s.svc.cluster.local:%s", targetNamespace, serverPort)
+	} else {
+		cachedTektonResultsHost = fmt.Sprintf("tekton-results-api-service.openshift-pipelines.svc.cluster.local:%s", serverPort)
+	}
+	return cachedTektonResultsHost, nil
+}
+
+func GetTektonResults(r *http.Request, user *auth.User, dynamicClient *dynamic.DynamicClient) (common.DevConsoleCommonResponse, error) {
+	var request TektonResultsRequest
+	err := json.NewDecoder(r.Body).Decode(&request)
+	if err != nil {
+		return common.DevConsoleCommonResponse{}, fmt.Errorf("failed to parse request: %v", err)
+	}
+
+	TEKTON_RESULTS_HOST, err := getTRHost(r, dynamicClient)
+	if err != nil {
+		return common.DevConsoleCommonResponse{}, fmt.Errorf("failed to get TektonResults host: %v", err)
+	}
+
+	TEKTON_RESULTS_URL := fmt.Sprintf("https://%s/apis/results.tekton.dev/v1alpha2/parents/%s/results/-/records?%s", TEKTON_RESULTS_HOST, request.SearchNamespace, request.SearchParams)
```

Comment:
Do we need to worry about encoding any of these path elements or are they definitely URL safe?

Reply:
@Lucifergene If these values are coming from the HTTP request query string, we can't trust them. We don't know for sure that this request is coming from console, and someone could craft parameters values that tack additional items to the query string. For example,

```
searchNamespace = "foo&some_other_param=some_other_value"
```

While if this is just a query endpoint, maybe in practice it couldn't cause damage, but it's easy to guard against. Let's properly encode the query values here and elsewhere in this PR so it's never a problem.